### PR TITLE
[WFLY-6684]: JMS Bridge should display statistics about messages that have been processed.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding RFE description.

Jira: https://issues.jboss.org/browse/WFLY-6684

### CODE REVIEW

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+=== Other Interested Projects
+
+== Requirements
+
+* The messageCount and abortedMessageCount attributes of ActiveMQ Artemis JMS Bridge configuration must be readable via Wildfly management model.
+
+== Implementation Plan
+
+The `message-count` and `aborted-message-count` attributes will be added to the jms bridge resource.
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: `message-count` is pretty clear to me. I however don't know how messages in `aborted-message-count` will be count (under which circumstances the message is added here)? Could you please help me?

Also, please add their type and if they are just read-only (I guess they are).

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+CLI to set the attribute:
+
+----
+/subsystem=messaging-activemq/jms-bridge=bridge:read-attribute(name=message-count)
+/subsystem=messaging-activemq/jms-bridge=bridge:read-attribute(name=aborted-message-count)
+----
+
+== Test Plan
+
+* Changes in Wildfly management model will be covered in WF testsuite.
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: What kinds of tests are planned? Functional, integration or just unit tests?

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+* https://issues.jboss.org/browse/ENTMQBR-1004
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mnovak@redhat.com[Miroslav Novak]
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: I guess that's me now - `jkasik`

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-6684
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/ENTMQBR-1004
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: Can you please leave a note how it is related?

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+== Requirements
+
+* The messageCount and abortedMessageCount attributes of ActiveMQ Artemis JMS Bridge configuration must be readable via Wildfly management model.
+
+== Implementation Plan
+
+The `message-count` and `aborted-message-count` attributes will be added to the jms bridge resource.
+
+CLI to set the attribute:
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: If those attributes are just read-only there is no way of setting them I guess...

File: messaging/WFLY-6684_jms_bridge_statistics.adoc (hunk 2 above)
Comment (ehsavoie) [reply messaging/WFLY-6684_jms_bridge_statistics.adoc]: Simple integration test

File: messaging/WFLY-6684_jms_bridge_statistics.adoc (hunk 4 above)
Comment (ehsavoie) [reply messaging/WFLY-6684_jms_bridge_statistics.adoc]: This is the feature creating those statistics

File: messaging/WFLY-6684_jms_bridge_statistics.adoc (hunk 5 above)
Comment (ehsavoie) [reply messaging/WFLY-6684_jms_bridge_statistics.adoc]: fixed

File: messaging/WFLY-6684_jms_bridge_statistics.adoc (hunk 3 above)
Comment (ehsavoie) [reply messaging/WFLY-6684_jms_bridge_statistics.adoc]: fixed

File: messaging/WFLY-6684_jms_bridge_statistics.adoc (hunk 1 above)
Comment (ehsavoie) [reply messaging/WFLY-6684_jms_bridge_statistics.adoc]: fixed

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+CLI to read the attributes:
+
+----
+/subsystem=messaging-activemq/jms-bridge=bridge:read-attribute(name=message-count)
+/subsystem=messaging-activemq/jms-bridge=bridge:read-attribute(name=aborted-message-count)
+----
+
+== Test Plan
+
+* Changes in Wildfly management model will be covered in WF testsuite by a simple integration tes: the test only covers the integration with Artemis aka reading the values from Artemis.
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: *integration test

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+* https://issues.apache.org/jira/browse/ARTEMIS-561
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kasik]
+
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: Missing Testing by section here, see my previous comment...

File: messaging/WFLY-6684_jms_bridge_statistics.adoc [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+An Artemis JMS Bridge consumes messages from a source JMS destination, and send them to a target JMS destination.
+Typically either the source or the target destinations are on different servers. The bridge can also be used to bridge messages from other non Artemis
+JMS servers, as long as they are JMS 1.1 compliant.
+The goal of this RFE is to display statistics on the number of processed messages and how many were aborted/rolled back.
```
Comment (honza-kasik) [inline messaging/WFLY-6684_jms_bridge_statistics.adoc]: Please also add a note here saying, that subject of this RFE is only displaying statistics from Artemis.

**Comment by ehsavoie** [comment]
@jamezp looks like this was not merged :)

**Comment by jamezp** [comment]
Whoops yeah. Thanks for reminding me :)

**Pull request merged**